core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -76,14 +76,14 @@ public static String concatPath(Object base, Object... paths) throws IllegalArgu
     Preconditions.checkArgument(paths != null, "Failed to concatPath: a null set of paths");
     List<String> trimmedPathList = new ArrayList<>();
     String trimmedBase =
-        CharMatcher.is(AlluxioURI.SEPARATOR.charAt(0)).trimTrailingFrom(base.toString().trim());
+        CharMatcher.is(AlluxioURI.SEPARATOR.charAt(0)).trimTrailingFrom(base.toString());
```

Comment:
why is this change needed?

Reply:
By fixing this bug we now make the master throw error if a directory is not created correctly in UFS. This triggers the mkdir with space tests to fail because we trim the path before creating it on UFS. This line removed the trim so we can actually create directories with leading or trailing spaces on UFS.